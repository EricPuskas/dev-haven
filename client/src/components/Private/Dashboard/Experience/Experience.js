import React from "react";
import Moment from "react-moment";

const Experience = ({ experience, deleteExperience }) => {
  const experiences = experience.map(exp => (
    <tr key={exp._id}>
      <td>{exp.company}</td>
      <td className="hide-sm">{exp.title}</td>
      <td>
        <Moment format="YYYY MMM">{exp.from}</Moment>
        <span className="hide-sm"> - </span>
        {exp.to === null ? " Now" : <Moment format="YYYY MMM">{exp.to}</Moment>}
      </td>
      <td>
        <button
          className="btn btn-danger"
          onClick={() => deleteExperience(exp._id)}
        >
          <i className="fas fa-trash-alt" />
        </button>
      </td>
    </tr>
  ));
  return (
    <div className="Dashboard-exp-container">
      <h1> Experience Credentials </h1>
      <div
        style={{
          backgroundColor: "rgb(249, 249, 249"
        }}
      >
        <table className="table f4">
          <thead>
            <tr>
              <th>Company</th>
              <th className="hide-sm">Title </th>
              <th>Years </th>
              <th />
            </tr>
          </thead>
          <tbody>{experiences}</tbody>
        </table>
      </div>
    </div>
  );
};

export default Experience;
