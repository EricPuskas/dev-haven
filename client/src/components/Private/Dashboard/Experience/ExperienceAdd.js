import React, { useState } from "react";
import { connect } from "react-redux";
import { addExperience } from "../../../../actions/profile";
import { Link, withRouter } from "react-router-dom";
import Input from "../../../Common/Input";

const AddExperience = ({ errors, addExperience, history }) => {
  const { error } = errors;
  const [formData, setFormData] = useState({
    company: "",
    title: "",
    location: "",
    from: "",
    to: "",
    current: false,
    description: ""
  });

  const [toDateDisabled, toggleDisabled] = useState(false);
  const { company, title, location, from, to, current, description } = formData;
  const handleChange = e =>
    setFormData({ ...formData, [e.target.name]: e.target.value });
  return (
    <div className="ProfileNew">
      <h1>Add An Experience</h1>
      <p>
        Add any developer/programming positions that you have had in the past
      </p>
      <small>* = required field</small>
      <div className="ProfileNew-container">
        <div className="main-div">
          <form
            className="ProfileNew-form"
            onSubmit={e => {
              e.preventDefault();
              addExperience(formData, history);
            }}
          >
            <label>* Job Title</label>
            <Input
              placeholder="* Job Title"
              name="title"
              type="text"
              value={title}
              onChange={handleChange}
              error={error.title}
              error_fixed
            />
            <label>* Company</label>
            <Input
              placeholder="* Company"
              name="company"
              type="text"
              value={company}
              onChange={handleChange}
              error={error.company}
              error_fixed
            />
            <label>Location</label>
            <Input
              placeholder="Location"
              name="location"
              type="text"
              value={location}
              onChange={handleChange}
              error={error.location}
              error_fixed
            />
            <label>From Date</label>
            <Input
              name="from"
              type="date"
              value={from}
              onChange={handleChange}
              error={error.from}
              error_fixed
            />
            <div className="form-group">
              <p>
                <input
                  type="checkbox"
                  name="current"
                  checked={current}
                  value={current}
                  onChange={e => {
                    setFormData({ ...formData, current: !current });
                    toggleDisabled(!toDateDisabled);
                  }}
                />{" "}
                <small>Current Job</small>
              </p>
            </div>
            <label>To Date</label>
            <Input
              name="to"
              type="date"
              disabled={toDateDisabled ? "disabled" : ""}
              value={to}
              onChange={handleChange}
              error={error.to}
              error_fixed
            />
            <div className="form-group">
              <textarea
                name="description"
                cols="30"
                rows="5"
                placeholder="Job Description"
                value={description}
                onChange={handleChange}
              />
            </div>
            <div className="ProfileNew-controls">
              <button
                type="submit"
                className="btn btn-standard btn-submit"
                id="ProfileNew-submit-btn"
              >
                Submit
              </button>
              <Link className="btn btn-standard btn-go-back" to="/dashboard">
                Go Back
              </Link>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { addExperience }
)(withRouter(AddExperience));
