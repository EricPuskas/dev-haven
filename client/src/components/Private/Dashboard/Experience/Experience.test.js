import { shallow } from "enzyme";
import React from "react";
import Experience from "./Experience";

it("expect to render Experience component", () => {
  const mockExp = [
    {
      _id: 1,
      company: "Google",
      title: "Developer",
      from: "10/18/2018",
      to: null
    }
  ];

  expect(
    shallow(<Experience experience={mockExp} />).debug()
  ).toMatchSnapshot();
});
