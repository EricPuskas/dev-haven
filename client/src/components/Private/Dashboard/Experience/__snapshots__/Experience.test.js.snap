// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Experience component 1`] = `
"<div className=\\"Dashboard-exp-container\\">
  <h1>
     Experience Credentials 
  </h1>
  <div style={{...}}>
    <table className=\\"table f4\\">
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th className=\\"hide-sm\\">
            Title 
          </th>
          <th>
            Years 
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Google
          </td>
          <td className=\\"hide-sm\\">
            Developer
          </td>
          <td>
            <t format=\\"YYYY MMM\\" element={{...}} fromNow={false} toNow={false} calendar={false} ago={false} unix={false} utc={false} local={false} unit={{...}} withTitle={false} decimal={false} titleFormat=\\"\\" interval={60000} filter={[Function: filter]} onChange={[Function: onChange]}>
              10/18/2018
            </t>
            <span className=\\"hide-sm\\">
               - 
            </span>
             Now
          </td>
          <td>
            <button className=\\"btn btn-danger\\" onClick={[Function: onClick]}>
              <i className=\\"fas fa-trash-alt\\" />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>"
`;
